proxy/src/http/source.rs

```diff
@@ -296,7 +298,9 @@ mod handler {
     ) -> Result<impl Reply, Rejection> {
         let urn = project_urn.parse().map_err(Error::from)?;
```

Comment:
If we move the `urn` parsing into the `with_api` block we can actually avoid the `Error::from` which is sort of nice :) I did it in a few places but missed others.